ui/perfherder/compare/CompareTable.jsx

```diff
@@ -6,14 +6,127 @@ import { faHashtag } from '@fortawesome/free-solid-svg-icons';
 
 import { getHashBasedId } from '../helpers';
 import { hashFunction } from '../../helpers/utils';
+import { compareTableSort } from '../constants';
 
 import RetriggerButton from './RetriggerButton';
 import CompareTableRow from './CompareTableRow';
+import CompareSortButton from './CompareSortButton';
+
+export default class CompareTable extends React.Component {
+  sortTypes = {
+    [compareTableSort.ascending]: {
+      sortByString: (value) => (a, b) => a[value].localeCompare(b[value]),
+      sortByValue: (value) => (a, b) => a[value] - b[value],
+    },
+    [compareTableSort.descending]: {
+      sortByString: (value) => (a, b) => b[value].localeCompare(a[value]),
+      sortByValue: (value) => (a, b) => b[value] - a[value],
+    },
+  };
+
+  constructor(props) {
+    super(props);
+    const { data } = this.props;
+
+    this.state = {
+      data,
+      tableConfig: {
+        TestName: {
+          name: 'Test name',
+          sortValue: 'name',
+          currentSort: compareTableSort.default,
+        },
+        Base: {
+          name: 'Base',
+          sortValue: 'originalValue',
+          currentSort: compareTableSort.default,
+        },
+        Comparison: { name: 'Comparison' },
+        New: {
+          name: 'New',
+          sortValue: 'newValue',
+          currentSort: compareTableSort.default,
+        },
+        Delta: {
+          name: 'Delta',
+          sortValue: 'deltaPercentage',
+          currentSort: compareTableSort.default,
+        },
+        Magnitude: {
+          name: 'Magnitude of Difference',
+          sortValue: 'magnitude',
+          currentSort: compareTableSort.default,
+        },
+        Confidence: {
+          name: 'Confidence',
+          sortValue: 'confidence',
+          currentSort: compareTableSort.default,
+        },
+      },
+    };
+  }
+
+  componentDidUpdate = (prevProps) => {
+    const { data } = this.props;
+    const { tableConfig } = this.state;
+
+    if (data !== prevProps.data) {
+      Object.keys(tableConfig).forEach((key) => {
+        tableConfig[key].currentSort = compareTableSort.default;
+      });
+      // eslint-disable-next-line react/no-did-update-set-state
+      this.setState({ data, tableConfig });
+    }
+  };
+
+  getNextSort = (currentSort) => {
+    const { ascending, descending, default: defaultSort } = compareTableSort;
+    const nextSort = {
+      ascending: descending,
+      descending: defaultSort,
+      default: ascending,
+    };
+
+    return nextSort[currentSort];
+  };
+
+  sort = (sortValue, sortType, data) => {
+    const validData = data.filter((item) => item[sortValue]);
```

Comment:
According to this filter condition, we don't treat 0 as a valid value. My guess is: if the user does an ascending sort, he won't see the 0-valued rows 1st, rather at the end of the column.

If the `props.data` we receive defaults some of the values to 0 instead of `null`, then we need to look deeper into the code that's providing these values & correct it. It's weird that we have 0-value rows that aren't visible...